In [2]:
import numpy as np
import os
import sys
import torch

/home/carol/anaconda3/envs/anotherone/lib/python3.8/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


## Check dictionaries


In [1]:
params3 = {"path": "/home/carol/NaroNet-main/NaroNet-main/EXP2/", "PCL_embedding_dimensions": 256, "PCL_batch_size": 160, "PCL_epochs": 300, "PCL_patch_size": 32, "PCL_alpha_L": 1.3, "PCL_ZscoreNormalization": True,
"PCL_width_CNN": 2, "PCL_depth_CNN": 50, "experiment_Label": ["Response"], "num_samples_architecture_search": 100, "epochs": 20, "epoch": 0, "lr_decay_factor": 0.5, "lr_decay_step_size": 12, "weight_decay": 0.01,
"batch_size": 6, "lr": 0.001, "useOptimizer": "ADAM", "context_size": 15, "num_classes": 3, "MultiClass_Classification": 1, "showHowNetworkIsTraining": True, "visualizeClusters": True, "learnSupvsdClust": True, 
"recalculate": False, "folds": 13, "device": "cuda", "normalizeFeats": False, "normalizeCells": False, "Batch_Normalization": True, "normalizePercentile": False, "dataAugmentationPerc": 0.0001, "hiddens": 44,
"clusters1": 10, "clusters2": 9, "clusters3": 7, "LSTM": False, "GLORE": False, "Phenotypes": True, "DeepSimple": False, "isAttentionLayer": False, "ClusteringOrAttention": True, "1cell1cluster": False,
"dropoutRate": 0.2, "AttntnSparsenss": False, "attntnThreshold": 0, "GraphConvolution": "ResNet", "n-hops": 3, "modeltype": "SAGE", "ObjectiveCluster": True, "ReadoutFunction": False,
"NearestNeighborClassification": False, "NearestNeighborClassification_Lambda0": 1, "NearestNeighborClassification_Lambda1": 1, "NearestNeighborClassification_Lambda2": 1, "KinNearestNeighbors": 5,
"pearsonCoeffSUP": False, "pearsonCoeffUNSUP": False, "orthoColor": True, "orthoColor_Lambda0": 0.1, "orthoColor_Lambda1": 1e-05, "ortho": False, "ortho_Lambda0": 0.1, "ortho_Lambda1": 0, "ortho_Lambda2": 0,
"min_Cell_entropy": True, "min_Cell_entropy_Lambda0": 1, "min_Cell_entropy_Lambda1": 0.0001, "min_Cell_entropy_Lambda2": 0.01, "MinCut": True, "MinCut_Lambda0": 0, "MinCut_Lambda1": 0.1, "MinCut_Lambda2": 0.1,
"F-test": False, "Max_Pat_Entropy": False, "Max_Pat_Entropy_Lambda0": 0.0001, "Max_Pat_Entropy_Lambda1": 0.1, "Max_Pat_Entropy_Lambda2": 0.1, "UnsupContrast": False, "UnsupContrast_Lambda0": 0,
"UnsupContrast_Lambda1": 0, "UnsupContrast_Lambda2": 0, "Lasso_Feat_Selection": False, "Lasso_Feat_Selection_Lambda0": 0.1, "SupervisedLearning_Lambda0": 1, "SupervisedLearning_Lambda1": 1, 
"SupervisedLearning_Lambda2": 1, "SupervisedLearning_Lambda3": 1, "SupervisedLearning": True}

params2 = {'path': '/home/carol/NaroNet-main/NaroNet-main/EXP2/', 'PCL_embedding_dimensions': 256, 'PCL_batch_size': 160, 'PCL_epochs': 300, 'PCL_patch_size': 32,
'PCL_alpha_L': 1.3, 'PCL_ZscoreNormalization': True, 'PCL_width_CNN': 2, 'PCL_depth_CNN': 50, 'experiment_Label': ['Response'], 'num_samples_architecture_search': 50,
'epochs': 10, 'epoch': 0, 'lr_decay_factor': 0.5, 'lr_decay_step_size': 12, 'weight_decay': 0.01, 'batch_size': 6, 'lr': 0.00001, 'useOptimizer': 'ADAM', 'context_size': 15,
'num_classes': 3, 'MultiClass_Classification': 1, 'showHowNetworkIsTraining': False, 'visualizeClusters': True, 'learnSupvsdClust': True, 'recalculate': False, 'folds': 13,
'device': 'cuda', 'normalizeFeats': False, 'normalizeCells': False, 'Batch_Normalization': True, 'normalizePercentile': False, 'dataAugmentationPerc': 0.0001, 'hiddens': 44,
'clusters1': 10, 'clusters2': 9, 'clusters3': 7, 'LSTM': False, 'GLORE': False, 'Phenotypes': True, 'DeepSimple': False, 'isAttentionLayer': False,
'ClusteringOrAttention': True, '1cell1cluster': False, 'dropoutRate': 0.2, 'AttntnSparsenss': False, 'attntnThreshold': 0, 'GraphConvolution': 'ResNet', 'n-hops': 3, 
'modeltype': 'SAGE', 'ObjectiveCluster': True, 'ReadoutFunction': False, 'NearestNeighborClassification': False, 'NearestNeighborClassification_Lambda0': 1, 
'NearestNeighborClassification_Lambda1': 1, 'NearestNeighborClassification_Lambda2': 1, 'KinNearestNeighbors': 5, 'pearsonCoeffSUP': False, 'pearsonCoeffUNSUP': False,
'orthoColor': True, 'orthoColor_Lambda0': 0.1, 'orthoColor_Lambda1': 1e-05, 'ortho': False, 'ortho_Lambda0': 0.1, 'ortho_Lambda1': 0, 'ortho_Lambda2': 0,
'min_Cell_entropy': True, 'min_Cell_entropy_Lambda0': 1, 'min_Cell_entropy_Lambda1': 0.0001, 'min_Cell_entropy_Lambda2': 0.01, 'MinCut': True,
'MinCut_Lambda0': 0, 'MinCut_Lambda1': 0.1, 'MinCut_Lambda2': 0.1, 'F-test': False, 'Max_Pat_Entropy': False, 'Max_Pat_Entropy_Lambda0': 0.0001,
'Max_Pat_Entropy_Lambda1': 0.1, 'Max_Pat_Entropy_Lambda2': 0.1, 'UnsupContrast': False, 'UnsupContrast_Lambda0': 0, 'UnsupContrast_Lambda1': 0,
'UnsupContrast_Lambda2': 0, 'Lasso_Feat_Selection': False, 'Lasso_Feat_Selection_Lambda0': 0.1, 'SupervisedLearning_Lambda0': 1, 
'SupervisedLearning_Lambda1': 1, 'SupervisedLearning_Lambda2': 1, 'SupervisedLearning_Lambda3': 1, 'SupervisedLearning': True}

params4 = {"path": "/home/carol/NaroNet-main/NaroNet-main/EXP2/", "PCL_embedding_dimensions": 256, "PCL_batch_size": 160, "PCL_epochs": 300, "PCL_patch_size": 32, "PCL_alpha_L": 1.3, "PCL_ZscoreNormalization": True, "PCL_width_CNN": 2,
"PCL_depth_CNN": 50, "experiment_Label": ["Response"], "num_samples_architecture_search": 100, "epochs": 20, "epoch": 0, "lr_decay_factor": 0.5, "lr_decay_step_size": 12, "weight_decay": 0.01, "batch_size": 6, "lr": 0.001,
"useOptimizer": "ADAM", "context_size": 15, "num_classes": 3, "MultiClass_Classification": 1, "showHowNetworkIsTraining": False, "visualizeClusters": True, "learnSupvsdClust": True, "recalculate": False, "folds": 13, 
"device": "cuda", "normalizeFeats": False, "normalizeCells": False, "Batch_Normalization": True, "normalizePercentile": False, "dataAugmentationPerc": 0.0001, "hiddens": 44, "clusters1": 10, "clusters2": 9, "clusters3": 7,
"LSTM": False, "GLORE": False, "Phenotypes": True, "DeepSimple": False, "isAttentionLayer": False, "ClusteringOrAttention": True, "1cell1cluster": False, "dropoutRate": 0.2, "AttntnSparsenss": False, "attntnThreshold": 0,
"GraphConvolution": "ResNet", "n-hops": 3, "modeltype": "SAGE", "ObjectiveCluster": True, "ReadoutFunction": False, "NearestNeighborClassification": False, "NearestNeighborClassification_Lambda0": 1,
"NearestNeighborClassification_Lambda1": 1, "NearestNeighborClassification_Lambda2": 1, "KinNearestNeighbors": 5, "pearsonCoeffSUP": False, "pearsonCoeffUNSUP": False, "orthoColor": True, "orthoColor_Lambda0": 0.1,
"orthoColor_Lambda1": 1e-05, "ortho": False, "ortho_Lambda0": 0.1, "ortho_Lambda1": 0, "ortho_Lambda2": 0, "min_Cell_entropy": True, "min_Cell_entropy_Lambda0": 1, "min_Cell_entropy_Lambda1": 0.0001,
"min_Cell_entropy_Lambda2": 0.01, "MinCut": True, "MinCut_Lambda0": 0, "MinCut_Lambda1": 0.1, "MinCut_Lambda2": 0.1, "F-test": False, "Max_Pat_Entropy": False, "Max_Pat_Entropy_Lambda0": 0.0001,
"Max_Pat_Entropy_Lambda1": 0.1, "Max_Pat_Entropy_Lambda2": 0.1, "UnsupContrast": False, "UnsupContrast_Lambda0": 0, "UnsupContrast_Lambda1": 0, "UnsupContrast_Lambda2": 0, "Lasso_Feat_Selection": False,
"Lasso_Feat_Selection_Lambda0": 0.1, "SupervisedLearning_Lambda0": 1, "SupervisedLearning_Lambda1": 1, "SupervisedLearning_Lambda2": 1, "SupervisedLearning_Lambda3": 1, "SupervisedLearning": True}

In [2]:
print(params4 == params3)

False


In [5]:
differences = compare_dicts(params4, params3)
print(differences)

{'keys_only_in_dict1': set(), 'keys_only_in_dict2': set(), 'differing_values': {'showHowNetworkIsTraining': (False, True)}}


In [4]:
def compare_dicts(dict1, dict2):
    differences = {
        'keys_only_in_dict1': set(dict1.keys()) - set(dict2.keys()),
        'keys_only_in_dict2': set(dict2.keys()) - set(dict1.keys()),
        'differing_values': {}
    }

    for key in dict1.keys() & dict2.keys():
        if dict1[key] != dict2[key]:
            differences['differing_values'][key] = (dict1[key], dict2[key])

    return differences

## Check generated graphs

In [12]:
# Load the .pt file
file_path = '/home/carol/NaroNet-main/NaroNet-main/Endometrial_POLE/NaroNet/POLE Mutation_Copy number variation_MSI Status_Tumour Type/Subject_graphs/data_0_0.pt'
file_path = '/home/carol/NaroNet-main/NaroNet-main/Lung_cancer/NaroNet/Response/Subject_graphs/data_1_0.pt'
loaded_data = torch.load(file_path)

data_type = type(loaded_data)
print("Type of the loaded data:", data_type)

# Assuming loaded_data is an instance of torch_geometric.data.Data
print("Node feature matrix shape:", loaded_data.x.shape if loaded_data.x is not None else "Not present")
print("Graph connectivity shape:", loaded_data.edge_index.shape if loaded_data.edge_index is not None else "Not present")
print("Edge feature matrix shape:", loaded_data.edge_attr)
print("Target shape:", loaded_data.y if loaded_data.y is not None else "Not present")


Type of the loaded data: <class 'torch_geometric.data.data.Data'>
Node feature matrix shape: torch.Size([33605, 262])
Graph connectivity shape: torch.Size([2, 167253])
Edge feature matrix shape: None
Target shape: [0]


Node feature matrix shape: torch.Size([11625, 263])
Graph connectivity shape: torch.Size([2, 57682])
Edge feature matrix shape: None
Target shape: ['Mutated', 'High', 'Stable', 'Endometria carcinoma ']


## Adding eps 

In [ ]:
# Create two NumPy arrays
array1 = np.array([[1, 2, 3], [4, 5, 6]])
array2 = np.array([[7, 8, 9], [10, 11, 12]])

# Specify the file name
filename = 'my_arrays.npz'

# Save the arrays in a single file
np.savez(filename, array1=array1, array2=array2)

In [14]:
# Assuming denom is your array
denom = np.array([[[1, 0, 1, 1, 1, 1]]])
denom = denom.astype(np.float64)
# Define a small value, epsilon
epsilon = 1e-10

# Add epsilon to only the zero elements
denom[denom == 0] += epsilon

# Now denom no longer contains any zero values
print(denom)

[[[1.e+00 1.e-10 1.e+00 1.e+00 1.e+00 1.e+00]]]


In [11]:
denom = np.array([[[1, 0, 1, 4, 2, 1]]])
print(np.any(denom.tolist() == 0)) 

False


In [12]:
denom[denom == 0] += sys.float_info.epsilon
print(denom)

UFuncTypeError: Cannot cast ufunc 'add' output from dtype('float64') to dtype('int64') with casting rule 'same_kind'

## Check parameters dict

In [10]:
dict1 = {'path': '/home/carol/NaroNet-main/NaroNet-main/Lung_cancer/', 'PCL_embedding_dimensions': 256, 'PCL_batch_size': 160, 'PCL_epochs': 300, 'PCL_patch_size': 32,
        'PCL_alpha_L': 1.3, 'PCL_ZscoreNormalization': True, 'PCL_width_CNN': 2, 'PCL_depth_CNN': 50, 'experiment_Label': ['Response'], 'num_samples_architecture_search': 50,
        'epochs': 10, 'epoch': 0, 'lr_decay_factor': 0.5, 'lr_decay_step_size': 12, 'weight_decay': 0.01, 'batch_size': 6, 'lr': 0.001, 'useOptimizer': 'ADAM', 'context_size': 15,
        'num_classes': 3, 'MultiClass_Classification': 1, 'showHowNetworkIsTraining': True, 'visualizeClusters': True, 'learnSupvsdClust': True, 'recalculate': False, 'folds': 13,
        'device': 'cuda', 'normalizeFeats': False, 'normalizeCells': False, 'Batch_Normalization': True, 'normalizePercentile': False, 'dataAugmentationPerc': 0.0001, 'hiddens': 44,
        'clusters1': 10, 'clusters2': 9, 'clusters3': 7, 'LSTM': False, 'GLORE': False, 'Phenotypes': True, 'DeepSimple': False, 'isAttentionLayer': False,
        'ClusteringOrAttention': True, '1cell1cluster': False, 'dropoutRate': 0.2, 'AttntnSparsenss': False, 'attntnThreshold': 0, 'GraphConvolution': 'ResNet', 'n-hops': 3, 
        'modeltype': 'SAGE', 'ObjectiveCluster': True, 'ReadoutFunction': False, 'NearestNeighborClassification': False, 'NearestNeighborClassification_Lambda0': 1, 
        'NearestNeighborClassification_Lambda1': 1, 'NearestNeighborClassification_Lambda2': 1, 'KinNearestNeighbors': 5, 'pearsonCoeffSUP': False, 'pearsonCoeffUNSUP': False,
        'orthoColor': True, 'orthoColor_Lambda0': 0.1, 'orthoColor_Lambda1': 1e-05, 'ortho': False, 'ortho_Lambda0': 0.1, 'ortho_Lambda1': 0, 'ortho_Lambda2': 0,
        'min_Cell_entropy': True, 'min_Cell_entropy_Lambda0': 1, 'min_Cell_entropy_Lambda1': 0.0001, 'min_Cell_entropy_Lambda2': 0.01, 'MinCut': True,
        'MinCut_Lambda0': 0, 'MinCut_Lambda1': 0.1, 'MinCut_Lambda2': 0.1, 'F-test': False, 'Max_Pat_Entropy': False, 'Max_Pat_Entropy_Lambda0': 0.0001,
        'Max_Pat_Entropy_Lambda1': 0.1, 'Max_Pat_Entropy_Lambda2': 0.1, 'UnsupContrast': False, 'UnsupContrast_Lambda0': 0, 'UnsupContrast_Lambda1': 0,
        'UnsupContrast_Lambda2': 0, 'Lasso_Feat_Selection': False, 'Lasso_Feat_Selection_Lambda0': 0.1, 'SupervisedLearning_Lambda0': 1, 
        'SupervisedLearning_Lambda1': 1, 'SupervisedLearning_Lambda2': 1, 'SupervisedLearning_Lambda3': 1, 'SupervisedLearning': True}

dict2 = {"path": "/home/carol/NaroNet-main/NaroNet-main/Lung_cancer/", "PCL_embedding_dimensions": 256, "PCL_batch_size": 160, "PCL_epochs": 300, "PCL_patch_size": 32,
        "PCL_alpha_L": 1.3, "PCL_ZscoreNormalization": True, "PCL_width_CNN": 2, "PCL_depth_CNN": 50, "experiment_Label": ["Response"], "num_samples_architecture_search": 50,
        "epochs": 10, "epoch": 0, "lr_decay_factor": 0.5, "lr_decay_step_size": 12, "weight_decay": 0.01, "batch_size": 6, "lr": 0.001, "useOptimizer": "ADAM", "context_size": 15,
        "num_classes": 3, "MultiClass_Classification": 1, "showHowNetworkIsTraining": True, "visualizeClusters": True, "learnSupvsdClust": True, "recalculate": False,
        "folds": 13, "device": "cuda", "normalizeFeats": False, "normalizeCells": False, "Batch_Normalization": True, "normalizePercentile": False, 
        "dataAugmentationPerc": 0.0001, "hiddens": 44, "clusters1": 10, "clusters2": 9, "clusters3": 7, "LSTM": False, "GLORE": False, "Phenotypes": True,
        "DeepSimple": False, "isAttentionLayer": False, "ClusteringOrAttention": True, "1cell1cluster": False, "dropoutRate": 0.2, "AttntnSparsenss": False, 
        "attntnThreshold": 0, "GraphConvolution": "ResNet", "n-hops": 3, "modeltype": "SAGE", "ObjectiveCluster": True, "ReadoutFunction": False,
        "NearestNeighborClassification": False, "NearestNeighborClassification_Lambda0": 1, "NearestNeighborClassification_Lambda1": 1,
        "NearestNeighborClassification_Lambda2": 1, "KinNearestNeighbors": 5, "pearsonCoeffSUP": False, "pearsonCoeffUNSUP": False, "orthoColor": True,
        "orthoColor_Lambda0": 0.1, "orthoColor_Lambda1": 1e-05, "ortho": False, "ortho_Lambda0": 0.1, "ortho_Lambda1": 0, "ortho_Lambda2": 0, "min_Cell_entropy": True,
        "min_Cell_entropy_Lambda0": 1, "min_Cell_entropy_Lambda1": 0.0001, "min_Cell_entropy_Lambda2": 0.01, "MinCut": True, "MinCut_Lambda0": 0,
        "MinCut_Lambda1": 0.1, "MinCut_Lambda2": 0.1, "F-test": False, "Max_Pat_Entropy": False, "Max_Pat_Entropy_Lambda0": 0.0001, "Max_Pat_Entropy_Lambda1": 0.1,
        "Max_Pat_Entropy_Lambda2": 0.1, "UnsupContrast": False, "UnsupContrast_Lambda0": 0, "UnsupContrast_Lambda1": 0, "UnsupContrast_Lambda2": 0, 
        "Lasso_Feat_Selection": False, "Lasso_Feat_Selection_Lambda0": 0.1, "SupervisedLearning_Lambda0": 1, "SupervisedLearning_Lambda1": 1,
        "SupervisedLearning_Lambda2": 1, "SupervisedLearning_Lambda3": 1, "SupervisedLearning": True}

In [11]:
for key in dict1.keys():
    if dict1[key]!=dict2[key]:
        print(str(key) + ' ' + str(dict1[key]) + ' ' + str(dict2[key]))

In [12]:
print(dict1==dict2)

True


## Checking number of patches  

In [13]:
path = "/home/carol/NaroNet-main/NaroNet-main/Lung_cancer/Patch_Contrastive_Learning/Image_Patch_Representation/h13_10873_tl1.npy"
print(np.shape(np.load(path)))

(9270, 264)


In [14]:
path = "/home/carol/NaroNet-main/NaroNet-main/Lung_cancer/Patch_Contrastive_Learning/Image_Patch_Representation" 
for filename in os.listdir(path):
    npar = np.load(path+'/'+filename)
    print(filename + str(np.shape(npar)))

h17_6794_tl2.npy(31000, 264)
h17_6226_tl4.npy(32800, 264)
h16_16787_tl2.npy(19580, 264)
h17_6226_tl2.npy(32800, 264)
h16_15119_tl1.npy(32480, 264)
h16_11096_tl1.npy(24038, 264)
h16_19144_tl2.npy(23800, 264)
h16_15767_tl3.npy(36516, 264)
h16_28421_tl2.npy(31416, 264)
h16_19144_tl4.npy(24252, 264)
h16_16787_tl1.npy(22784, 264)
h16_7700_tl2.npy(35076, 264)
h16_11096_tl5.npy(7636, 264)
h16_11096_tl3.npy(31866, 264)
h17_6794_tl1.npy(20800, 264)
h16_12101_tl1.npy(33488, 264)
h16_11096_tl2.npy(27813, 264)
h16_15119_tl2.npy(24160, 264)
h16_19144_tl1.npy(39720, 264)
h17_6226_tl1.npy(32800, 264)
h14_28759_tl2.npy(31020, 264)
h16_28421_tl4.npy(31416, 264)
h16_28421_tl3.npy(31416, 264)
h13_10873_tl2.npy(33605, 264)
h16_11096_tl4.npy(41514, 264)
h16_15767_tl1.npy(33464, 264)
h16_19144_tl3.npy(21109, 264)
h16_28421_tl1.npy(31416, 264)
h16_19144_tl7.npy(24252, 264)
h16_12101_tl2.npy(22650, 264)
h16_7700_tl1.npy(30336, 264)
h16_15767_tl2.npy(30222, 264)
h13_10873_tl1.npy(9270, 264)
h16_9373_tl1.npy(33

In [15]:
path = "/home/carol/NaroNet-main/Datasets/PCL_outputs/Patch_Contrastive_Learning_ps22/Image_Patch_Representation"
for filename in os.listdir(path):
    npar = np.load(path+'/'+filename)
    print(filename + str(np.shape(npar)))

h16_12101_tl1.npy(70819, 136)
h16_19144_tl7.npy(51324, 136)
h16_9373_tl1.npy(70788, 136)


In [4]:
#checking path array sizes for A.patch representations and B. preprocesed images

path = "/home/carol/NaroNet-main/Datasets/IPR/Image_Patch_Representation_ps22"

for filename in os.listdir(path):
    npar = np.load(path+'/'+filename)
    print(filename + str(np.shape(npar)))

h17_6794_tl2.npy(31000, 264)
h17_6226_tl4.npy(32800, 264)
h16_16787_tl2.npy(19580, 264)
h17_6226_tl2.npy(32800, 264)
h16_11096_tl1.npy(24038, 264)
h16_19144_tl2.npy(23800, 264)
h16_28421_tl2.npy(31416, 264)
h16_16787_tl1.npy(22784, 264)
h16_7700_tl2.npy(35076, 264)
h16_11096_tl3.npy(31866, 264)
h16_11096_tl2.npy(27813, 264)
h16_19144_tl1.npy(39720, 264)
h16_28421_tl4.npy(31416, 264)
h13_10873_tl2.npy(33605, 264)
h16_28421_tl1.npy(31416, 264)
h16_12101_tl2.npy(22650, 264)
h16_7700_tl1.npy(30336, 264)
h16_15767_tl2.npy(30222, 264)
h16_9373_tl1.npy(33320, 264)
h13_10873_tl3.npy(2950, 264)
h16_19144_tl6.npy(24252, 264)
h16_9373_tl2.npy(29750, 264)
h17_6226_tl3.npy(32800, 264)
h16_19144_tl5.npy(24252, 264)


In [4]:
#comparing if image patch representation vectors are the same on two different iterations
path1 = "/home/carol/NaroNet-main/NaroNet-main/Lung_cancer/Patch_Contrastive_Learning/Image_Patch_Representation/h13_10873_tl2.npy"
path2 = "/home/carol/NaroNet-main/Datasets/Image_Patch_Representation_ps22/h13_10873_tl2.npy"
path3 = "/home/carol/NaroNet-main/Datasets/Image_Patch_Representation_ps22/run2_ps32/h13_10873_tl2.npy"
array1 = np.load(path1)
array2 = np.load(path2)
array3 = np.load(path3)
print(np.shape(array1))
print(np.shape(array2))
print(np.shape(array3))

(33605, 264)
(33605, 264)
(33605, 264)


In [5]:
path4 = "/home/carol/NaroNet-main/NaroNet-main/Lung_cancer/Patch_Contrastive_Learning/Preprocessed_Images/h13_10873_tl2.npy"
array4 = np.load(path4)
print(np.shape(array4))

(7550, 4600, 6)


## Decimal checks

In [32]:
rounded_arr1_2dp = np.around(array3, decimals=2)
rounded_arr2_2dp = np.around(array2, decimals=2)
are_equal_2dp = np.array_equal(rounded_arr1_2dp, rounded_arr2_2dp)

# Compare for third decimal place
rounded_arr1_3dp = np.around(array3, decimals=3)
rounded_arr2_3dp = np.around(array2, decimals=3)
are_equal_3dp = np.array_equal(rounded_arr1_3dp, rounded_arr2_3dp)

# Output the results
print("Are the arrays equal up to the second decimal? ", are_equal_2dp)
print("Are the arrays equal up to the third decimal? ", are_equal_3dp)

Are the arrays equal up to the second decimal?  False
Are the arrays equal up to the third decimal?  False


In [33]:
# Compare for third decimal place
rounded_arr1_4dp = np.around(array3, decimals=4)
rounded_arr2_4dp = np.around(array2, decimals=4)
are_equal_4dp = np.array_equal(rounded_arr1_3dp, rounded_arr2_3dp)
print("Are the arrays equal up to the second decimal? ", are_equal_4dp)

Are the arrays equal up to the second decimal?  False


In [34]:
differences_2dp = np.sum(rounded_arr1_2dp != rounded_arr2_2dp)
print("Differences for second decimal place: ", differences_2dp)

Differences for second decimal place:  66269


In [35]:
differences_3dp = np.sum(rounded_arr1_3dp != rounded_arr2_3dp)
print("Differences for second decimal place: ", differences_3dp)

Differences for second decimal place:  662834


In [25]:
print(array3[0])

[ 1.60000000e+01  1.60000000e+01 -1.72941767e-01 -1.72943636e-01
 -1.72941865e-01 -1.72943731e-01 -1.72941915e-01 -1.72943788e-01
 -2.52729368e+00  1.60067332e+00 -4.81508851e-01 -6.12240434e-02
  5.60304642e-01  7.08035290e-01  1.22876811e+00  1.63058090e+00
  1.50733495e+00 -8.78988206e-02 -6.57963634e-01  2.96522677e-01
  2.42388630e+00 -1.06474221e+00 -5.93141079e-01  9.74468529e-01
  5.37092328e-01 -9.18477595e-01 -4.40133125e-01 -8.22815955e-01
  7.55571604e-01 -5.67771673e-01  1.93880707e-01  1.62770963e+00
  1.12464786e+00 -3.50794196e-01 -2.27988958e-01 -8.97854924e-01
 -9.25166607e-02 -6.99304819e-01  1.72314033e-01  1.33716881e+00
  1.18558228e-01 -3.15756381e-01 -2.03897285e+00  1.43722379e+00
  1.03845394e+00  4.47935820e-01  2.77847886e-01  4.48035896e-01
  1.76331565e-01 -8.19253683e-01 -1.16527128e+00 -2.21640611e+00
  5.63945413e-01  6.22443497e-01  9.02707696e-01 -7.73391604e-01
 -6.44636512e-01  7.25969732e-01 -7.84394622e-01 -1.39271450e+00
 -1.40775287e+00  2.01210

In [26]:
print(array2[0])

[ 1.60000000e+01  1.60000000e+01 -1.72941767e-01 -1.72943636e-01
 -1.72941865e-01 -1.72943731e-01 -1.72941915e-01 -1.72943788e-01
 -2.52690983e+00  1.60044646e+00 -4.81510282e-01 -6.10085130e-02
  5.60173512e-01  7.07803786e-01  1.22861016e+00  1.63035679e+00
  1.50737119e+00 -8.79045427e-02 -6.58199191e-01  2.96498060e-01
  2.42369556e+00 -1.06482673e+00 -5.93218923e-01  9.74289298e-01
  5.37066281e-01 -9.17933762e-01 -4.40269023e-01 -8.22759807e-01
  7.55840063e-01 -5.67345083e-01  1.94079846e-01  1.62742746e+00
  1.12512815e+00 -3.50810617e-01 -2.27823615e-01 -8.98178339e-01
 -9.26491022e-02 -6.99446559e-01  1.72344878e-01  1.33732104e+00
  1.18570805e-01 -3.15785468e-01 -2.03931022e+00  1.43711209e+00
  1.03853202e+00  4.47996140e-01  2.77933061e-01  4.47711200e-01
  1.76133856e-01 -8.19282413e-01 -1.16544974e+00 -2.21607351e+00
  5.64109802e-01  6.22300565e-01  9.02819753e-01 -7.73557067e-01
 -6.44600213e-01  7.26174116e-01 -7.84283519e-01 -1.39259505e+00
 -1.40767920e+00  2.01170

## Statistical analysis

Image embeddings should of size 128 according to paper.
Check if all 264 positions are non-zero. 
Check what's the difference between patch rep and prep images

In [4]:
def explore(image_data):
    
    print("Image shape:", image_data.shape)

    mean_value = np.mean(image_data)
    std_value = np.std(image_data)
    min_value = np.min(image_data)
    max_value = np.max(image_data)

    print("Mean:", mean_value)
    print("Standard Deviation:", std_value)
    print("Minimum:", min_value)
    print("Maximum:", max_value)

In [5]:
explore(npar[1])

Image shape: (264,)
Mean: 0.20679186495261342
Standard Deviation: 3.20183238504033
Minimum: -2.2506768703460693
Maximum: 48.0
